### PR TITLE
Add federated data centre support for NDACC

### PR DESCRIPTION
**Pull request by victoriarspada**
* clear records before loading
* new files in woudc-data-registry/woudc_data_registry/peer
* station variations text file 

Test file:  https://www-air.larc.nasa.gov/pub/NDACC/PUBLIC/listings/WOUDC_csv.txt
Note that the datetime strings are formatted incorrectly in this file, and need to add trailing Z, as per the error message below:
`[2021-07-02T15:03:20Z] ERROR - time data '2005-12-01T00:00:00' does not match format '%Y-%m-%dT%H:%M:%SZ'`

I made this fix with the following file: 
[fixdatetime.txt](https://github.com/woudc/woudc-data-registry/files/6793559/fixdatetime.txt)

### CODE REVIEW

File: woudc_data_registry/models.py [hunk 1]
```diff
@@ -1137,8 +1137,9 @@ class PeerDataRecord(base):
     station_name_id = Column(String,
                              ForeignKey('station_names.station_name_id'),
                              nullable=False)
-    stn_type_enum = Enum('land', name='stn_type')
-    station_type = Column(stn_type_enum, nullable=False, default='land')
+    stn_type_enum = Enum('STN','SHP','land', name='stn_type')
```
Comment (tomkralidis) [inline woudc_data_registry/models.py:1140]: Use just `land`, for broad interoperability of peer data records.

File: woudc_data_registry/peer/file_indices_extractor.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import os, csv, logging, psycopg2
```
Comment (tomkralidis) [inline woudc_data_registry/peer/file_indices_extractor.py:1]: Add a source code header ([example](https://github.com/woudc/woudc-data-registry/blob/master/woudc_data_registry/log.py#L1-L44))

File: woudc_data_registry/peer/file_indices_extractor.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from urllib.request import urlopen
+
+LOGGER = logging.getLogger(__name__)
+
+Process=dict(
```
Comment (tomkralidis) [inline woudc_data_registry/peer/file_indices_extractor.py:6]: `s/Process/PROCESS/`

File: woudc_data_registry/peer/file_indices_extractor.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+def get_station_metadata(name, stations, variations):
+    """Look up station metadata given a station name."""
+    key = name 
+    if key is not None:
+        key = name.lower()
+    if key in stations:
+        return stations[key]
+    elif variations is not None:
+        msg = 'Station metadata not found for the station name: %s' % name
```
Comment (tomkralidis) [inline woudc_data_registry/peer/file_indices_extractor.py:48]: change to `msg = 'Station metadata not found for the station name: {}'.format(name)`.  Apply to all string interpolation in this PR.

**Review by tomkralidis — CHANGES_REQUESTED** [review]
Great work!  Also run the Python files through flake8.

File: woudc_data_registry/models.py (hunk 1 above)
Comment (victoriarspada) [reply woudc_data_registry/models.py:1140]: For the time being, should I set station_type to 'land' for NDACC files if the corresponding WOUDC station is type 'STN'? As of now I'm querying the database for this property for NDACC files.

Comment (tomkralidis) [reply woudc_data_registry/models.py]: Yes, `land` is good for now (so if you derive `STN` from the WOUDC db query, set as `land` in the peer record table).

**Pull request merged**